File: h2go-chaincodes/models/request.go
package models

import "encoding/json"

type Request struct {
	DocType     string        `json:"docType"`
	RequestID   string        `json:"requestId"`
	ProducerID  string        `json:"producerId"`
	AssetType   AssetType     `json:"assetType"`
	Amount      int64         `json:"amount"`
	Status      RequestStatus `json:"status"`
	ApproverID  string        `json:"approverId"`
	Reason      string        `json:"reason"`
	GdOs        []GdO         `json:"gdos"`
	CreatedAt   string        `json:"createdAt"`
	ProcessedAt string        `json:"processedAt"`
}

func (r *Request) UnmarshalJSON(data []byte) error {
	// Usamos un alias para evitar recursión infinita
	type Alias Request
	aux := &struct {
		*Alias
	}{
		Alias: (*Alias)(r),
	}

	// Deserializar normalmente
	if err := json.Unmarshal(data, aux); err != nil {
		return err
	}

	// Corregir GdOs si es nil
	if r.GdOs == nil {
		r.GdOs = []GdO{}
	}

	return nil
}

File: frontend/src/app/types/request.ts
import { AssetType } from "./assetType";
import { GdO } from "./gdo";
import { RequestStatus } from "./requestStatus";

export interface Request {
  docType: string;
  requestId: string;
  producerId: string;
  assetType: AssetType | string;
  amount: number;
  status: RequestStatus | string;
  approverId: string;
  reason: string;
  gdos: GdO[];
  createdAt: string;
  processedAt: string;
}
